crates/flox/src/commands/package.rs

```diff
@@ -369,10 +374,58 @@ impl Publish {
         info!("done!");
 
         info!("Checking binary can be downloaded from {substituter_url}...");
-        publish
-            .check_substituter(substituter_url)
-            .await
-            .context("Binary cannot be downloaded")?;
+
+        let mut retry_count = None;
+        let found_upstream = loop {
+            let check_result = publish.check_substituter(substituter_url.clone()).await;
+
+            if check_result.is_ok() {
+                break true;
+            }
+
+            let err = check_result.unwrap_err();
+            warn!("Unable to find binary at {substituter_url}: {err}");
+
+            // in first round:
+            // if we can prompt, confirm that we should retry
+            // else              assume yes, retry {self.retry} times
+            if retry_count.is_none() && Dialog::can_prompt() {
+                let help_message = Some(format!(
+                    "flox query {substituter_url} {} more times; waiting 5 seconds inbetween.",
+                    self.retry
+                ));
+                let confirm = Dialog {
+                    message: "Keep looking for binary?",
+                    help_message: help_message.as_deref(),
+                    typed: Confirm {
+                        default: Some(true),
+                    },
+                };
+
+                if !confirm.prompt().await? {
+                    bail!("aborted publish!");
+                }
+            }
+
+            // get the current try count
+            let current_try = retry_count.get_or_insert(1);
```

Comment:
Not sure if it's idiomatic and just new to me, but I thought making this an optional was pretty weird

Reply:
I rather check for is_none rather than is 0.. I agree it's personal preference, if you guys hate it i'll change it - or use some other `first_try: bool` for this.